frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -361,7 +366,7 @@ export const StorageClassEncryption: React.FC<ProvisionerProps> = ({
   React.useEffect(() => {
     if (isKmsSupported && !_.isEmpty(csiConfigMap)) {
       const serviceNames = Object.keys(csiConfigMap?.data);
-      const kmsData = JSON.parse(csiConfigMap?.data[serviceNames[serviceNames.length - 1]]);
+      const kmsData = JSON.parse(csiConfigMap?.data[getMostRecentKMS(serviceNames)]);
```

Comment:
Used regex instead of sorting:
```
const serviceNames = Object.keys(csiConfigMap?.data).sort(new Intl.Collator(undefined, {numeric:true}).compare);
const kmsData = JSON.parse(csiConfigMap?.data[serviceNames[serviceNames.length - 1]]);
```
As, I guess, it will be comparatively more optimised. (taking O(n) right now, will take O(nlogn) with sorting.

Reply:
You fixed it partially, need to fix this in ```StorageClassEncryptionKMSID``` also

Reply:
ohh... okay... ack...